inference-engine/src/inference_engine/compilation_context.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
+#include <string>
+#include <vector>
+#include <map>
+#include <xml_parse_utils.h>
+
+#include "ie_itt.hpp"
+#include "cpp_interfaces/exception2status.hpp"
+#include "transformations/serialize.hpp"
+#include "cnn_network_ngraph_impl.hpp"
+#include "cpp/ie_cnn_network.h"
+#include "details/ie_exception.hpp"
+
+#include "ngraph/pass/pass.hpp"
+#include "ngraph/variant.hpp"
+#include "ngraph/function.hpp"
+#include "ngraph/opsets/opset6.hpp"
+
+#ifdef WIN32
+#define stat _stat
+#endif
+
+namespace InferenceEngine {
+
+struct NetworkCompilationContext final {
+    static std::string calculateFileInfo(const std::string& filePath) {
+        std::string res;
+        struct stat result;
+        size_t seed {};
+        seed = hash_combine(seed, filePath);
+        if (stat(filePath.c_str(), &result) == 0) {
+            seed = hash_combine(seed, result.st_mtime);
+            seed = hash_combine(seed, result.st_size);
+        }
+        return std::to_string(seed);
+    }
+
+    static std::string computeHash(const CNNNetwork& network,
+                                   const std::map<std::string, std::string>& compileOptions) {
+        OV_ITT_SCOPED_TASK(itt::domains::IE_LT, "NetworkCompilationContext::computeHash - CNN");
+        std::stringstream xmlFile;
+        std::stringstream binFile;
+
+        // 1. Serialize
+        CNNNetwork networkCopy = network; // TODO: it is not a real copy
+
+        // TODO: add extensions (custom ops) to serializer
+        ngraph::pass::Serialize serializer(xmlFile, binFile,
+            ngraph::pass::Serialize::Version::IR_V10);
+        serializer.run_on_function(networkCopy.getFunction());
+
+        // 2. Compute hash on serialized data and options
+        size_t seed {};
+        seed = hash_combine(seed, xmlFile.str());
+        seed = hash_combine(seed, binFile.str());
+        for (const auto& kvp : compileOptions) {
+            seed = hash_combine(seed, kvp.first + kvp.second);
+        }
```

Comment:
I see that you have removed hashing of runtime information like affinity, primitivesPriority, preprocessing info and so on. Any reason to do this?

BTW, this information is not serialized.